docker/docker.ts

```diff
@@ -175,36 +180,76 @@ async function buildAndPushImageWorkerAsync(
     };
 
     // If the container specified a cacheFrom parameter, first set up the cached stages.
-    let cacheFrom: Promise<string[] | undefined>;
+    let cacheFrom: string[] | undefined;
     if (typeof pathOrBuild !== "string" && pathOrBuild && pathOrBuild.cacheFrom) {
-        // NOTE: we pull the promise out of the repository URL s.t. we can observe whether or not it exists. Were we
-        // to instead hang an apply off of the raw Input<>, we would never end up running the pull if the repository
-        // had not yet been created.
+        // NOTE: we pull the promise out of the repository URL s.t. we can observe whether or not it
+        // exists. Were we to instead hang an apply off of the raw Input<>, we would never end up
+        // running the pull if the repository had not yet been created.
         const cacheFromParam = typeof pathOrBuild.cacheFrom === "boolean" ? {} : pathOrBuild.cacheFrom;
-        cacheFrom = pullCacheAsync(imageName, cacheFromParam, login, repositoryUrl, logResource);
-    } else {
-        cacheFrom = Promise.resolve(undefined);
+        cacheFrom = await pullCacheAsync(imageName, cacheFromParam, login, repositoryUrl, logResource);
     }
 
     // First build the image.
-    const buildResult = await buildImageAsync(imageName, pathOrBuild, logResource, cacheFrom);
+    const { imageId, stages } = await buildImageAsync(imageName, pathOrBuild, logResource, cacheFrom);
+
+    if (!repositoryUrl) {
+        throw new ResourceError("Expected repository URL to be defined during push", logResource);
+    }
```

Comment:
Why do we need this?  The parameter is marked non-optional.

Reply:
I was preserving/moving this from existing code.  But you're right that it shouldn't be necessary in either case.  will remove.